### PR TITLE
Parking Slot edit action added

### PR DESCRIPTION
**Pull request by SardorBozorov**

### CODE REVIEW

File: ParkingZone/Models/ParkingSlotVMs/EditVM.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    public uint Number { get; set; }
+    [Required]
+    public bool IsAvailable { get; set; }
+    [Required]
+    public ParkingSlotCategory Category { get; set; }
+    [Required]
+    public long ParkingZoneId { get; set; }
+    public long Id { get; set; }
+    public EditVM()
+    { }
```
Comment (iamurtazo) [inline ParkingZone/Models/ParkingSlotVMs/EditVM.cs]:
```suggestion
public EditVm() {} lets keep it like this in line 18
```

File: Parking_Zone.Test/ModelValidationtests/ParkingSlots/EditVMTest.cs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+    [Theory]
+    [MemberData(nameof(TestData))]
+    public void GivenItemToBeValidated_WhenEditSlot_ThenValidationIsPerformed
+        (long id, uint number, ParkingSlotCategory category, bool isAvailable, long parkingZoneId, bool expectedValidation)
+    {
+        //Arrange
+        EditVM listItemVM = new EditVM()
+        {
+
```
Comment (iamurtazo) [inline Parking_Zone.Test/ModelValidationtests/ParkingSlots/EditVMTest.cs]: space

File: ParkingZone/Areas/Admin/Controllers/ParkingSlotController.cs [hunk 3]
```diff
@@ -58,4 +58,39 @@ public IActionResult Create(CreateVM createVM)
...
+    [HttpGet]
+    public IActionResult Edit(long id)
+    {
+        var slot = _parkingSlotService.GetById(id);
+
+        if (slot is null)
+            return NotFound();
+        var editVM = new EditVM(slot);
+        return View(editVM);
+    }
```
Comment (dilmurod-rozikov) [inline ParkingZone/Areas/Admin/Controllers/ParkingSlotController.cs:72]: put a space after closing braces

File: ParkingZone/Models/ParkingSlotVMs/EditVM.cs [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    public EditVM(ParkingSlot parkingSlot)
+    {
+        Number = parkingSlot.Number;
+        Category = parkingSlot.Category;
+        IsAvailable = parkingSlot.IsAvailable;
+        ParkingZoneId = parkingSlot.ParkingZoneId;
+        Id = parkingSlot.Id;
+    }
+    public ParkingSlot MapToModel(ParkingSlot Vm)
```
Comment (dilmurod-rozikov) [inline ParkingZone/Models/ParkingSlotVMs/EditVM.cs]: public ParkingSlot MapToModel(ParkingSlot Vm)
```suggestion
    public ParkingSlot MapToModel(ParkingSlot slot)
```

File: Parking_Zone.Test/Controllers/ParkingSlotControllerTest.cs [hunk 5]
```diff
@@ -123,4 +123,124 @@ public void GivenCreateVM_WhenCreatePostIsCalled_ThenModelStateIsTrueAndReturnsR
...
+
+            //Assert
+            Assert.NotNull(result);
+            Assert.IsType<NotFoundResult>(result);
+        }
+
+        [Fact]
+        public void GivenEditVMAndParkingSlotId_WhenEditPostIsCalled_ThenReturnNotFoundIfSlotIsNull()
+        {
+        //Arrange
```
Comment (dilmurod-rozikov) [inline Parking_Zone.Test/Controllers/ParkingSlotControllerTest.cs]:
```suggestion
               //Arrange
```
put a space

**Review by dilmurod-rozikov — CHANGES_REQUESTED** [review]

File: Parking_Zone.Test/Controllers/ParkingSlotControllerTest.cs [hunk 6]
```diff
@@ -123,4 +123,124 @@ public void GivenCreateVM_WhenCreatePostIsCalled_ThenModelStateIsTrueAndReturnsR
...
+            var result = _parkingSlotController.Edit(_parkingSlot[0].Id);
+
+            //Assert
+            Assert.NotNull(result);
+            Assert.IsType<NotFoundResult>(result);
+            _parkingSlotServiceMoq.Verify(x => x.GetById(_parkingSlot[0].Id), Times.Once);
+        }
+
+        [Fact]
+        public void GivenEditVMAndParkingSlotId_WhenEditPostIsCalled_ThenReturnNotFoundIfIdAndSlotIdDoesNotMatch()
```
Comment (SamDotNetDev) [inline Parking_Zone.Test/Controllers/ParkingSlotControllerTest.cs]: NamingConvention error 
```suggestion
        public void GivenEditVMAndParkingSlotId_WhenEditPostIsCalled_ThenIdAndSlotIdDoNotMatchAndReturnsNotFound()
```

File: Parking_Zone.Test/Controllers/ParkingSlotControllerTest.cs [hunk 7]
```diff
@@ -123,4 +123,124 @@ public void GivenCreateVM_WhenCreatePostIsCalled_ThenModelStateIsTrueAndReturnsR
         _parkingSlotServiceMoq.Verify(x => x.Create(It.IsAny<ParkingSlot>()), Times.Once);
     }
     #endregion
+
+     #region Edit
+        [Fact]
+        public void GivenParkingSlotId_WhenEditGetIsCalled_ThenReturnViewResult()
```
Comment (SamDotNetDev) [inline Parking_Zone.Test/Controllers/ParkingSlotControllerTest.cs]:
```suggestion
        public void GivenParkingSlotId_WhenEditIsCalled_ThenReturnsViewResult()
```

File: Parking_Zone.Test/Controllers/ParkingSlotControllerTest.cs [hunk 8]
```diff
@@ -123,4 +123,124 @@ public void GivenCreateVM_WhenCreatePostIsCalled_ThenModelStateIsTrueAndReturnsR
...
+
+            //Assert
+            var model = Assert.IsType<ViewResult>(result).Model;
+            Assert.NotNull(result);
+            Assert.Equal(JsonSerializer.Serialize(model), JsonSerializer.Serialize(expectedEditVM));
+            _parkingSlotServiceMoq.Verify(x => x.GetById(_parkingSlot[0].Id), Times.Once);
+        }
+
+        [Fact]
+        public void GivenParkingSlotId_WhenEditGetIsCalled_ThenReturnNotFound()
```
Comment (SamDotNetDev) [inline Parking_Zone.Test/Controllers/ParkingSlotControllerTest.cs]: Returns

File: Parking_Zone.Test/Controllers/ParkingSlotControllerTest.cs [hunk 9]
```diff
@@ -123,4 +123,124 @@ public void GivenCreateVM_WhenCreatePostIsCalled_ThenModelStateIsTrueAndReturnsR
...
+            //Act
+            var result = _parkingSlotController.Edit(editVM, _parkingSlot[0].Id);
+
+            //Assert
+            Assert.NotNull(result);
+            Assert.IsType<NotFoundResult>(result);
+        }
+
+        [Fact]
+        public void GivenEditVMAndParkingSlotId_WhenEditPostIsCalled_ThenReturnNotFoundIfSlotIsNull()
```
Comment (SamDotNetDev) [inline Parking_Zone.Test/Controllers/ParkingSlotControllerTest.cs]: Same fix naming

**Review by SamDotNetDev — CHANGES_REQUESTED** [review]

**Review by dilmurod-rozikov — APPROVED** [review]

**Review by abdullokhtolibjonov — APPROVED** [review]

**Pull request merged**